src/tree/updater_gpu_hist.cu

```diff
@@ -1328,23 +1329,47 @@ inline void DeviceShard<GradientSumT>::InitCompressedData(
 
 template <typename GradientSumT>
 inline void DeviceShard<GradientSumT>::CreateHistIndices(
-    const SparsePage& row_batch, size_t row_stride, int null_gidx_value) {
+    const SparsePage &row_batch,
+    const common::HistCutMatrix &hmat,
+    const RowStateOnDevice &device_row_state,
+    int rows_per_batch) {
+  // Has any been allocated for me in this batch?
+  if (!device_row_state.batch_n_rows) return;
+
+  size_t row_stride = this->ellpack_matrix.row_stride;
+
+  const auto &offset_vec = row_batch.offset.ConstHostVector();
+  /*! \brief row offset in SparsePage (the input data). */
+  thrust::device_vector<size_t> row_ptrs(device_row_state.batch_n_rows+1);
+  thrust::copy(
+    offset_vec.data() + device_row_state.row_offset,
+    offset_vec.data() + device_row_state.row_offset + device_row_state.batch_n_rows + 1,
```

Comment:
I would put a check here to make sure we don't read off the end of offset_vec.

Reply:
will do...